packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -127,9 +127,14 @@ export function pasteHandler( { HTML = '', plainText = '', mode = 'AUTO', tagNam
 	// First of all, strip any meta tags.
 	HTML = HTML.replace( /<meta[^>]+>/, '' );
 
-	// If we detect block delimiters, parse entirely as blocks.
-	if ( mode !== 'INLINE' && HTML.indexOf( '<!-- wp:' ) !== -1 ) {
-		return parseWithGrammar( HTML );
+	// If we detect block delimiters in HTML, parse entirely as blocks.
+	if ( mode !== 'INLINE' ) {
+		// Check plain text if there is no HTML.
+		const content = HTML ? HTML : plainText;
```

Comment:
Should this apply for just the `parseWithGrammar` consideration, or also for the rest of `rawHandling`'s treatment of where it currently considers "HTML" ?

Reply:
We have separate logic for that: https://github.com/WordPress/gutenberg/pull/14536/files#diff-7703746fe73380c92b8bfe2370aab646R155

We also want to run it through the Markdown converter in other cases.